File: 프로그래머스/Lv.0/팩토리얼/120848.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i=1; i<=n; i++) {
            if (n < factorial(i)) break;
            answer = i;
        }
        
        return answer;
    }
    
    int factorial(int num) {
        if (num == 1) return 1;
        return num * factorial(num - 1);
    }
}

File: 프로그래머스/Lv.0/팩토리얼/120848.js
function solution(n) {
    let answer = 0;
    
    for (let i=1; i<=n; i++) {
        if (n < factorial(i)) break;
        answer = i;
    }
    
    return answer;
}

function factorial(num) {
    if (num === 1) return 1;
    return num * factorial(num - 1);
}
